# Preliminary analysis on TERMINET SCHN data

In [2]:
import pandas as pd
from scipy.stats import pearsonr

In [3]:
df = pd.read_csv("./SOE_28-03-2022.xlsx - Sheet2.csv")

In [5]:
df[:20]

logID           DATE LOCAL        NAME  DESC       VAL          QF SOURCE
0   13805  3/28/22 12:41     N     CPU_USE   NaN  77.86260  0x00001000    PLC
1   13808  3/28/22 12:41     N   FAIL_CONF   NaN   0.00000  0x00001000    PLC
2   13809  3/28/22 12:41     N    FAIL_RTU   NaN   0.00000  0x00001000    PLC
3   13806  3/28/22 12:41     N     MEM_USE   NaN  62.66120  0x00001000    PLC
4   13804  3/28/22 12:41     N       PS1_V   NaN   5.38353  0x00001000    PLC
5   13810  3/28/22 12:41     N  RTU_HEALTH   NaN   3.00000  0x00001000    PLC
6   13807  3/28/22 12:41     N        TEMP   NaN  31.12500  0x00001000    PLC
7   13791  3/28/22 12:21     N     CPU_USE   NaN  76.67980  0x00001000    PLC
8   13798  3/28/22 12:21     N     CPU_USE   NaN  77.73820  0x00001000    PLC
9   13794  3/28/22 12:21     N   FAIL_CONF   NaN   0.00000  0x00001000    PLC
10  13801  3/28/22 12:21     N   FAIL_CONF   NaN   0.00000  0x00001000    PLC
11  13795  3/28/22 12:21     N    FAIL_RTU   NaN   0.00000  0x00001000    PLC
12  13802  3/28/22 12:21     N    FAIL_RTU   NaN   0.00000  0x00001000    PLC
13  13792  3/28/22 12:21     N     MEM_USE   NaN  62.68540  0x00001000    PLC
14  13799  3/28/22 12:21     N     MEM_USE   NaN  62.68480  0x00001000    PLC
15  13790  3/28/22 12:21     N       PS1_V   NaN   5.38353  0x00001000    PLC
16  13797  3/28/22 12:21     N       PS1_V   NaN   5.38453  0x00001000    PLC
17  13796  3/28/22 12:21     N  RTU_HEALTH   NaN   1.00000  0x00001000    PLC
18  13803  3/28/22 12:21     N  RTU_HEALTH   NaN   3.00000  0x00001000    PLC
19  13793  3/28/22 12:21     N        TEMP   NaN  31.18700  0x00001000    PLC

In [3]:
df["NAME"].unique()

array(['CPU_USE', 'FAIL_CONF', 'FAIL_RTU', 'MEM_USE', 'PS1_V',
       'RTU_HEALTH', 'TEMP'], dtype=object)

In [5]:
CPUs = df[df["NAME"] == "CPU_USE"]["VAL"].dropna()[:-1]
FAIL_CONFs = df[df["NAME"] == "FAIL_CONF"]["VAL"].dropna()
FAIL_RTUs = df[df["NAME"] == "FAIL_RTU"]["VAL"].dropna()
MEM_USEs = df[df["NAME"] == "MEM_USE"]["VAL"].dropna()
PS1_Vs = df[df["NAME"] == "PS1_V"]["VAL"].dropna()
RTU_HEALTHs = df[df["NAME"] == "RTU_HEALTH"]["VAL"].dropna()
TEMPs = df[df["NAME"] == "TEMP"]["VAL"].dropna()

In [6]:
print("len(CPUs):", len(CPUs))
print("len(FAIL_CONFs):", len(FAIL_CONFs))
print("len(FAIL_RTUs):", len(FAIL_RTUs))
print("len(MEM_USEs):", len(MEM_USEs))
print("len(PS1_Vs):", len(PS1_Vs))
print("len(RTU_HEALTHs):", len(RTU_HEALTHs))
print("len(TEMPs):", len(TEMPs))

len(CPUs): 377
len(FAIL_CONFs): 377
len(FAIL_RTUs): 377
len(MEM_USEs): 377
len(PS1_Vs): 377
len(RTU_HEALTHs): 377
len(TEMPs): 377


# ERROR distribution

In [7]:
print("FAIL_CONFs.unique", FAIL_CONFs.unique())
print("FAIL_RTUs.unique", FAIL_RTUs.unique())
print("RTU_HEALTHs.unique", RTU_HEALTHs.unique())

FAIL_CONFs.unique [0.]
FAIL_RTUs.unique [0.]
RTU_HEALTHs.unique [3. 1. 2.]


---------------> NB: there are no errors in the dataset!!!

# Correlations


## CPU

In [98]:
print("pearsonr(CPUs, TEMPs): ", pearsonr(CPUs, TEMPs)[1])
print("pearsonr(CPUs, MEM_USEs): ", pearsonr(CPUs, MEM_USEs)[1])
print("pearsonr(CPUs, PS1_Vs): ", pearsonr(CPUs, PS1_Vs)[1])
print("pearsonr(CPUs, RTU_HEALTHs): ", pearsonr(CPUs, RTU_HEALTHs)[1])

pearsonr(CPUs, TEMPs):  0.2505619822815974
pearsonr(CPUs, MEM_USEs):  8.786607940359446e-36
pearsonr(CPUs, PS1_Vs):  0.33321768947957675
pearsonr(CPUs, RTU_HEALTHs):  0.12338897563973102


## MEMORY USAGE


In [99]:
print("pearsonr(MEM_USEs, TEMPs): ", pearsonr(MEM_USEs, TEMPs)[1])
print("pearsonr(MEM_USEs, PS1_Vs): ", pearsonr(MEM_USEs, PS1_Vs)[1])
print("pearsonr(MEM_USEs, RTU_HEALTHs): ", pearsonr(MEM_USEs, RTU_HEALTHs)[1])

pearsonr(MEM_USEs, TEMPs):  0.0009284286611734351
pearsonr(MEM_USEs, PS1_Vs):  0.5632157774318161
pearsonr(MEM_USEs, RTU_HEALTHs):  0.4720148305912593


## PS1_vs

In [100]:
print("pearsonr(PS1_Vs, TEMPs): ", pearsonr(PS1_Vs, TEMPs)[1])
print("pearsonr(PS1_Vs, RTU_HEALTHs): ", pearsonr(PS1_Vs, RTU_HEALTHs)[1])

pearsonr(PS1_Vs, TEMPs):  0.23315064103529948
pearsonr(PS1_Vs, RTU_HEALTHs):  0.7334088000294624


## Conclusions from a preliminary analysis
There is no correlation with errors, since there are not errors in the dataset.
There is a weak corellation between:
- PS1_Vs and MEM_USEs (0.56)
- PS1_Vs and RTU_HEALTHs (0.73)